utils/GRI_bed_filter/version1_gene-only/gri_hg19_v1.2.R

```diff
@@ -0,0 +1,167 @@
+#GRI.V1=protein_coding gene + HGMD mutations (50 flanking bp) + Cinvar mutations (50 flanking bp)
+
+#library(GenomicRanges)
+library(dplyr)
+library(VariantAnnotation) # https://bioconductor.org/packages/release/bioc/vignettes/VariantAnnotation/inst/doc/VariantAnnotation.html#variant-call-format-vcf-files
+library(data.table)
+# # #check file md5
+# file_path <- "clinvar.vcf.gz"
+# md5_checksum <- digest::digest(file_path, algo = "md5", file = TRUE)
+# md5file<- readLines('clinvar.vcf.gz.md5')
+# md5_expected<-strsplit(md5file[1], " ")[[1]][1]
+# print(md5_checksum==md5_expected)
+# 
+# # uncompress .gz files
+# folder_path <- getwd()
+# gz_files <- list.files(path = folder_path, pattern = "\\.gz$", full.names = TRUE)
+# for (file in gz_files) {
+#   R.utils::gunzip(file, remove = FALSE) # Set remove = TRUE if you want to delete the .gz file after uncompressing
+# }
+
+# define some functions
+coverage_bed <- function(bed_file, sequence_report='sequence_report.tsv'){
+  hg.seq<-readr::read_tsv(sequence_report)
+  # bed<-readr::read_tsv('exome_filter.bed', col_names = FALSE)
+  ##calculate hg seq size
+  seq.24<-hg.seq[c(1:24),]
+  seq.size<-sum(seq.24$`Seq length`)
+  ##calculate bed size
+  bed.size<-sum(bed_file[,4])
+  ##get percentage
+  size.per<-bed.size/seq.size
+  print(paste(paste0(size.per*100,"%"),bed.size,seq.size))
+}
+
+# 1. Use gene info from GENCODE (hg19) for GRI ----------------------------
+#read gtf file from gencode: https://www.gencodegenes.org/human/release_46lift37.html
+gtf_file <- "gencode.v46lift37.basic.annotation.gtf"
+gtf_data <- rtracklayer::import(gtf_file)
```

Comment:
You can directly download put the link, isn't it?

Reply:
yep, I plan to modify the code later to retrieve data from the link directly